pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -777,3 +778,82 @@ func (m *cgroupManagerImpl) GetResourceStats(name CgroupName) (*ResourceStats, e
 	}
 	return toResourceStats(stats), nil
 }
+
+// Get the memory limit in bytes applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupMemoryConfig(name CgroupName) (uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
+	stats, err := getStatsSupportedSubsystems(cgroupPaths)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get stats supported cgroup subsystems for cgroup %v: %v", name, err)
+	}
+	return stats.MemoryStats.Usage.Limit, nil
+}
+
+// Get the cpu quota, cpu period, and cpu shares applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupCpuConfig(name CgroupName) (int64, uint64, uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
```

Comment:
I'd encourage you to take a look at https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/fscommon/utils.go, which handles some of the additional corner cases that comes with reading cgroup files.

Reply:
Yes, I had gone through this. But I missed TrimSpace check between GetCgroupParamString +  ParseInt below.

Reply:
GetCgroupParamString has Trimspace check, and max/ParseUint is not applicable here since we are looking for -1, a valid value in this case. Is there something else I missed?

Reply:
Nothing missed necessarily.  IMO using the helper functions they provide seems easier than reimplementing them ourselves.  But its up to you.

Reply:
Oh yea.. these are merely wrapper functions providing a uniform looking Getter/Setter interface and they are consuming those helpers from ```import github.com/opencontainers/runc/libcontainer/cgroups/fscommon``` . These helpers came in recently i think .. don't remember them being around when I implemented the first prototype.